docs/docs/contributing/code/guidelines.mdx

```diff
@@ -0,0 +1,33 @@
+# General guidelines
+
+Here are some things to keep in mind for all types of contributions:
+
+- Follow the ["fork and pull request"](https://docs.github.com/en/get-started/exploring-projects-on-github/contributing-to-a-project) workflow. Please do not try to push directly to this repo unless you are a maintainer.
+- Use the checked-in pull request template when opening pull requests. Note related issues and tag relevant maintainers.
+- Ensure your PR passes formatting, linting, and testing checks before requesting a review.
+  - If you would like comments or feedback on your current progress, you can instead mark your PR as a draft.
```

Comment:
don't know if we need to encourage this. high level ideas should be discussed in issues/discussions and implementations should still pass CI. so far policy has been to not review prs until they're out of draft